v4/newrelic/application.go

```diff
@@ -28,7 +30,10 @@ func (app *Application) StartTransaction(name string) *Transaction {
 		rootSpan: s,
 		thread: &thread{
 			currentSpan: s,
+			spanCount:   1,
 		},
+		app:  app,
```

Comment:
Should this be a pointer? I'm not sure, so could you explain your choice here so I can learn something?